File: install/get_achievements.py
#!/bin/env python

import sys
from xml.dom import minidom

if __name__ == "__main__":
    if len(sys.argv) < 2:
        exit("Usage: %s <achievements.xml>" % (sys.argv[0]))
    
    filename = sys.argv[1]
    xmldoc = minidom.parse(filename)
    itemlist = xmldoc.getElementsByTagName('achievement')

    print("INSERT INTO `v2_achievements` (`id`,`name`) VALUES")
    for s in itemlist:
        if s == itemlist[-1]:
            sep = ";"
        else:
            sep = ","
        s_escaped = s.attributes['title'].value.replace("'", "''")
        print("    (%s, '%s')%s" % (s.attributes['id'].value, s_escaped, sep))

File: install/get-achievements.php
<?php
# validate data
if (php_sapi_name() !== "cli") exit("Not in CLI Mode");
if ($argc < 2) exit(sprintf("Usage: php %s <achievements.xml>" . PHP_EOL, $argv[0]));
$filename = $argv[1];
if (!file_exists($filename)) exit(sprintf("File '%s' does not exist" . PHP_EOL, $filename));

$dom = new DOMDocument();
$dom->load($filename);
$achievements = $dom->getElementsByTagName("achievement");

echo "INSERT INTO `v3_achievements` (`id`, `name`) VALUES" . PHP_EOL;
foreach ($achievements as $i => $achievement)
{
    /** @var DOMElement $achievement */
    echo sprintf(
        "    (%s, '%s')%s" . PHP_EOL,
        $achievement->getAttribute("id"),
        str_replace("'", "''", $achievement->getAttribute("name")),
        $i == ($achievements->length - 1) ? ';' : ','
    );
}
